project2-beat-that/script.js

```diff
@@ -1,4 +1,90 @@
+// Roll dice
+let rolldice = function () {
+  let num = Math.ceil(Math.random() * 6);
+  return num;
+};
+
+// global variables
+let gamemode = "p1";
+let number1 = 0;
+let number2 = 0;
+let total1 = 0;
+let total2 = 0;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  let message = "";
+  // when restart button is click, input for output box will become restart
+  if (input == "restart") {
+    gamemode = "p1";
+    number1 = 0;
+    number2 = 0;
+    total1 = 0;
+    total2 = 0;
+    return "Welcome to Beat That! Player 1, roll your dice!";
+  }
+  // when restart button is click, input for score box will become resetscore
+  if (input == "resetscore") {
+    message = "Leaderboard" + "<br>" + "Player 1: 0" + "<br>" + "Player 2: 0";
+    return message;
+  }
+  // score box will display current score
+  if (input == "score") {
+    message = "Leaderboard" + "<br>";
+    if (total1 > total2) {
+      message =
+        message + `Player 1: ${total1}` + "<br>" + `Player 2: ${total2}`;
+    } else {
+      message =
+        message + `Player 2: ${total2}` + "<br>" + `Player 1: ${total1}`;
+    }
+    return message;
+  }
```

Comment:
As mentioned above , this is the part that can be a separate function so that the main function is shorter and easier to read!